File: advanced-se6/src/main/java/com/turing/advancedse6/tdd/Stack.java
package com.turing.advancedse6.tdd;

public class Stack {

	private int[] items;
	private int size= 0;
	static final int MAX_SIZE =10;
	int top = -1;
	
	public Stack()
	{
		this.items = new int[MAX_SIZE];
	}
	public int getSize() {
		return size;
	}

	public void push(int item) {
		this.items[++this.top] = item;
		this.size++;
	}

	public int peek() {
		
		return this.items[this.top];
	}
	public int pop() {
		
		this.size --;
		return this.items[this.top--];
	}

}

File: code/stack.js

/**
 * 
stack {
    items:[],
    top:-1
}
 */
function push(stack,ele)//stack obj literal
{
    stack.items[++stack.top]= ele;
}


function pop(stack)
{
    let ele= stack.items[stack.top--];
    return ele;
}

let stack1={
    items:[],
    top: -1
};
push(stack1,10);
push(stack1,20);

console.log(pop(stack1));
console.log(pop(stack1));